curator-x-async/src/main/java/org/apache/curator/x/async/details/AsyncCuratorFrameworkImpl.java

```diff
@@ -151,6 +153,13 @@ public AsyncRemoveWatchesBuilder removeWatches()
         return new AsyncRemoveWatchesBuilderImpl(client, filters);
     }
 
+    @Override
+    public AsyncWatchBuilder addWatch()
+    {
+        Preconditions.checkState(Compatibility.hasPersistentWatchers(), "addWatch() is not supported in the ZooKeeper library being used.");
```

Comment:
I would throw UnsupportedOperationException

Reply:
it's always been this way - I'd rather leave it for historical reasons